operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -102,8 +102,13 @@ public AbstractOperator(Vertx vertx, String kind, O resourceOperator, MetricsPro
         this.metrics = metrics;
 
         // Setup metrics
-        Tags metricTags = Tags.of(Tag.of("kind", kind()));
+        Tags metricTags;
 
+        if (selectorLabels != null) {
+            metricTags = Tags.of(Tag.of("kind", kind()), Tag.of("selector", selectorLabels.toSelectorString()));
+        } else {
+            metricTags = Tags.of(Tag.of("kind", kind()), Tag.of("selector", ""));
+        }
```

Comment:
what about?

```java
metricTags = Tags.of(Tag.of("kind", kind()), selectorLabels != null ? Tag.of("selector", selectorLabels.toSelectorString()) : Tag.of("selector", ""));
```
of for clarity
```java
String selectorValue = selectorLabels != null ? selectorLabels.toSelectorString() : "";
metricTags = Tags.of(Tag.of("kind", kind()), Tag.of("selector", selectorValue));
```

or have you tried what happens when you just put `null` as tag value instead of empty string?

Reply:
Yeah I tried it with null but it gives us error and cause the operator to fail,  since its a `not null` property. So the above workaround will work fine but we cant go with the null one